reports/models.py

```diff
@@ -0,0 +1,468 @@
+                # Annotate a new field that concatenates enrolment id and status
+                enrolment_id_status=models.functions.Concat(
+                    "_enrolment_id",
+                    models.Value(separator),
+                    "enrolment_status",
+                    output_field=models.CharField(),
+                ),
+            ).exclude(
+                enrolment_id_status__in=[
+                    "{}{}{}".format(
+                        entry["_enrolment_id"], separator, entry["enrolment_status"]
+                    )
+                    for entry in enrolment_ids_with_statuses
+                ]
+            )
+        )
+
+
+class EnrolmentReportManager(models.Manager):
+    def get_queryset(self):
+        return UnsyncedQuerySet(self.model, using=self._db)
+
+    def latest_sync(self):
+        """Returns latest updated_at, which represents the date of the last sync made"""
+        return self.get_queryset().latest_sync()
+
+    def unsynced(self, sync_from: datetime = None):
+        """Query unsynced enrolment report instances"""
+        return self.get_queryset().unsynced()
+
+    def has_missing(self) -> bool:
+        # TODO: Find n effective and more reliable query to check this.
+        return bool(self.count_missing())
+
+    def count_missing(self) -> int:
+        return (
+            occurrences_models.Enrolment.objects.all().count()
+            - self.model.objects.all().exclude(_enrolment__isnull=True).count()
+        )
+
+    def create_missing(
+        self, hydrate_linkedevents_event=False, sync_from: datetime = None
+    ):
+        """Create missing enrolment report instances."""
+        enrolments = self.model.enrolment_objects.missing(sync_from=sync_from)
+        reports = [self.model(_enrolment=enrolment) for enrolment in enrolments]
+        for report in reports:
+            report._rehydrate(hydrate_linkedevents_event=hydrate_linkedevents_event)
+
+        return self.bulk_create(reports)
+
+    def update_unsynced(
+        self, hydrate_linkedevents_event=False, sync_from: datetime = None
+    ):
+        """Sync existing unsynced enrolment report instances."""
+        # TODO: Do we need to update every field,
+        # or would it be ok to use bulk_update or queryset.update()?
+        reports = self.unsynced(sync_from=sync_from)
+        for report in reports:
+            report._rehydrate(hydrate_linkedevents_event=hydrate_linkedevents_event)
+            report.save()
+        return reports
+
+
+class EnrolmentReport(TimestampedModel):
+    """
+    Enrolment report model is used to offer some exportable data related to enrolments.
+    It stores the information about enrolments and their study groups,
+    the coordinates of the group unit position and the coordinates of
+    the event position to calculate the distance to event,
+    the event data like time frame of the event and te publisher,
+    capacity information, event keywords, etc.
+    """
+
+    """
```

Comment:
Just writing about my personal preferences here. I'm not entirely sure if there's is a strong convention regarding block comments. There's [this](https://www.python.org/dev/peps/pep-0008/#block-comments), but I'm not sure if that's a strong enough argument. I've usually only seen multi-line string used for strings and docstrings. For one or multi-line comments I personally always use `#`.

I'm not entirely sure if the Python will evaluate this string or would it be able to optimize and ignore it like comments beginning with `#`. :thinking:

Reply:
I changed them. Here is a conversation about them https://stackoverflow.com/a/7696966/784642